src/proc/syncer-processing-block.cpp

```diff
@@ -10,63 +10,116 @@
-            ss << "SYNCED: ";
-            auto composite = dynamic_cast<composite_frame*>(f.frame);
-            for (int i = 0; i < composite->get_embedded_frames_count(); i++)
+            auto opt = wopt.lock();
+            if( opt )
             {
-                auto matched = composite->get_frame(i);
-                ss << matched->get_stream()->get_stream_type() << " " << matched->get_frame_number() << ", "<<std::fixed<< matched->get_frame_timestamp()<<" ";
+                ++n_opts;
+                if( opt->is_true() )
+                {
+                    enabled = true;
+                    break;
+                }
             }
+        }
+        if( n_opts && ! enabled )
+        {
+            get_source().frame_ready( std::move( frame ) );
+            return;
+        }
 
-            LOG_DEBUG(ss.str());
-            env.matches.enqueue(std::move(f));
-        });
-
-        auto f = [&](frame_holder frame, synthetic_source_interface* source)
         {
-            // if the syncer is disabled passthrough the frame
-            bool enabled = false;
-            size_t n_opts = 0;
-            for( auto& wopt : _enable_opts )
+            std::lock_guard< std::mutex > lock( _mutex );
+
+            if( !_matcher )
             {
-                auto opt = wopt.lock();
-                if( opt )
-                {
-                    ++n_opts;
-                    if( opt->is_true() )
-                    {
-                        enabled = true;
-                        break;
-                    }
-                }
+                if(!create_matcher( frame, log ))
+                    get_source().frame_ready(std::move(frame));
             }
-            if( n_opts  &&  ! enabled )
+            else
             {
-                get_source().frame_ready( std::move( frame ) );
-                return;
+                auto env = syncronization_environment{ source, matches, log };
+                _matcher->dispatch(std::move(frame), env);
             }
+                
+        }
 
-            single_consumer_frame_queue<frame_holder> matches;
+        frame_holder f;
+        {
+            std::unique_lock< std::mutex > lock(_callback_mutex, std::try_to_lock);
+            if (!lock.owns_lock())
+                return;
 
+            while( matches.try_dequeue( &f ) )
             {
-                std::lock_guard<std::mutex> lock(_mutex);
-                _matcher->dispatch(std::move(frame), { source, matches });
+                get_source().frame_ready( std::move( f ) );
             }
+        }
+    };
 
-            frame_holder f;
-            while (matches.try_dequeue(&f))
-                get_source().frame_ready(std::move(f));
+    set_processing_callback( std::shared_ptr< rs2_frame_processor_callback >(
+        new internal_frame_processor_callback< decltype( f ) >( f ) ) );
+}
 
-        };
+bool syncer_process_unit::create_matcher( const frame_holder & frame, bool log )
```

Comment:
No need for `create_matcher` to get `log`

Reply:
done